### PR TITLE
Create django admin

### PR DESCRIPTION
**Pull request by mforner13**
This PR creates a django admin instance for the application.

**Configuring display of objects**
From feedback we had from future admin users on how they identify different objects, I have configured the appropriate values to appear in the list view for each model. Also, in order for the objects to be easily recognisable, I have added `__str__` methods to models which didn't have them.

**Logging into admin**
To avoid the need for admin users to have two sets of login details, I have overridden the `login()` method for the admin site. 

If a user is already authenticated (via SSO) and they are marked as `is_staff=True` they will be able to access the admin index page. If either of these are not true, they will be redirected to the app's index url, where if they're unauthenticated they will be redirected to SSO to login.

**Custom email domains field**
For the `GovDepartment.email_domains` field, as it is an `ArrayField`, I have created a form for this model, which specifies that the field type for email_domains is a [SplitArrayField](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/forms/#splitarrayfield). This provides a simpler UI for admin users adding new gov departments.

Screenshot 2021-05-18 at 14 01 36

**NB this PR also removes the `--collect-only` flag accidentally left in the GH action to run pytest.

### CODE REVIEW

File: defend_data_capture/supply_chains/test/test_admin.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from django.urls import reverse
+
+
+@pytest.mark.django_db
+class TestAdminLogin:
+    def test_login_unauthenticated(self):
+        """Test unauthenticated user is redirected to auth login url."""
+        client = Client()
+        response = client.get(reverse("admin:index"))
+        assert response.status_code == 302
```
Comment (mforner13) [inline defend_data_capture/supply_chains/test/test_admin.py:13]: I was unsure of how to check they've been redirected away from admin, since we don't want to be calling staff SSO in the tests (or have to set up mock sso just for this).

Any ideas would be v appreciated :)

Comment (NickFitz) [reply defend_data_capture/supply_chains/test/test_admin.py:13]: My test_monthly_update_views_user_journey.py tests check for being redirected to a specific URL., if that's what you need. For example:
```python
        expected_redirect_url = reverse(
            "monthly-update-info-edit",
            kwargs={
                "supply_chain_slug": strategic_action.supply_chain.slug,
                "strategic_action_slug": strategic_action.slug,
                "update_slug": monthly_update.slug,
            },
        )
        assert response.status_code == 302
        assert response.url == expected_redirect_url
```

File: defend_data_capture/defend_data_capture/settings.py [hunk 2]
```diff
@@ -28,11 +28,12 @@
 
 INSTALLED_APPS = [
     "authbroker_client",
-    "django.contrib.admin",
+    "django.contrib.admin.apps.SimpleAdminConfig",
     "django.contrib.auth",
     "django.contrib.contenttypes",
-    "django.contrib.sessions",
     "django.contrib.messages",
+    "django.contrib.postgres",
```
Comment (dev-digital) [inline defend_data_capture/defend_data_capture/settings.py:35]: Why was this required,  more for my knowledge

**Review by dev-digital — APPROVED** [review]
Looks good to me :+1:

File: defend_data_capture/supply_chains/templates/base.html [hunk 3]
```diff
@@ -41,7 +41,7 @@
                         Resilience tool
                     </a>
                     <span class="app-header-span">
-                        <p class="app-header-item">{{ request.user.first_name }} {{ request.user.last_name }} - {{ request.user.gov_department.name }}</p>
+                        <p class="app-header-item">{{ request.user }} - {{ request.user.gov_department.name }}</p>
```
[the remark this answers is not in the document]
Comment (mforner13) [reply defend_data_capture/supply_chains/templates/base.html:44]: So now that we have the `__str__` method for the user model which shows first name + last name, in the template, we can just refer to them as `request.user` to get the first name + last name, as it gets the string version automatically :)

File: defend_data_capture/defend_data_capture/settings.py (hunk 2 above)
Comment (mforner13) [reply defend_data_capture/defend_data_capture/settings.py:35]: Good question! It was to allow us to use the the Postgres SpitArrayField for the gov department email_domains field :) 
`from django.contrib.postgres.forms.array import SplitArrayField`

**Review by dev-digital — APPROVED** [review]

**Review by NickFitz — APPROVED** [review]
👍 +1
